repository_service_tuf/cli/artifact/add.py

```diff
@@ -0,0 +1,65 @@
+# SPDX-License-Identifier: MIT
+
+from click import Context
+
+from repository_service_tuf.cli import click, console
+from repository_service_tuf.cli.artifact import artifact
+from repository_service_tuf.helpers.api_client import URL, send_payload
+from repository_service_tuf.helpers.cli import (
+    create_artifact_payload_from_filepath,
+)
+
+
+@artifact.command()
+@click.argument(
+    "filepath",
+    type=click.Path(exists=True),
+    # Currently, this is required. If we support adding artifacts without
```

Comment:
nit:
We could use a following usage:

`rstuf artifact add file1.tar.gz`
It generates
```json
{
      "info": {
        "length": 11342,
        "hashes": {
          "blake2b-256": "716f6e863f744b9ac22c97ec7b76ea5f5908bc5b2f67c61510bfc4751384ea7a"
        },
        "custom": {
          "key": "value"
        }
      },
      "path": "file1.tar.gz"
    }
```

`rstuf artifact add file1.tar.gz --path download/projectA/`

```json
{
      "info": {
        "length": 11342,
        "hashes": {
          "blake2b-256": "716f6e863f744b9ac22c97ec7b76ea5f5908bc5b2f67c61510bfc4751384ea7a"
        },
        "custom": {
          "key": "value"
        }
      },
      "path": " download/projectA/file1.tar.gz"
    }
```

Reply:
I think I prefer the explicitness of how we currently have it. Plus in the example of:  `rstuf artifact add mypath/file-v1.0.3.tar.gz -p projectA/download/`

it would have to be: `projectA/download/file-v1.0.3.tar.gz`  (we cut off the `mypath`, which is easy to do but on the other hand the user has to make more assumptions or test it out if we don't document it better).

Reply:
In any case I tried the optional as well, you can see it here https://github.com/repository-service-tuf/repository-service-tuf-cli/tree/rstuf-artifact-add-command-optional-path.